### PR TITLE
[Keymap] add Hyena on idobo

### PR DESCRIPTION
**Pull request by frgomes**
Hyena is a compact layout designed for typists, polyglots, mathematicians, programmers and Emacs users.

Hyena for QMK is heavily influenced by its parent Hyena for ``X11`` systems, keeping the same aims. For historic information: [http://github.com/frgomes/hyena](http://github.com/frgomes/hyena).

## Description

This is an initial pull request which provides a pretty functional keymap. However, a lot of work is still necessary in order to get the QMK version of Hyena on pair with [Hyena for X11](https://github.com/frgomes/hyena). Currently, these features are missing in comparison with [Hyena for X11](https://github.com/frgomes/hyena):

* mathematical and typographical characters;
* accented letters for a number of European languages.

A programmable keyboard is far more flexible than a conventional keyboard, creating several possibilities and expanding the initial aims of Hyena for X11. However, it's important to keep compatibility between these two incarnations of the same idea, so that time and effort spent on one keyboard is preserved in case the user needs to migrate to another keyboard for whatever reason. A user acquainted to a programmable mechanical keyboard running Hyena should feel at home using Hyena straight on his/her laptop during a trip or a quick visit to a customer. The user should not be disrupted in case his/her powerful programmable is not within reach.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Initial implementation, containing a complete base layer.
* Support for Emacs on Linux and Windoze. In particular, supported was added in order to allow Emacs users circumvent the usual insanity involving arbitrary shortcuts on Windoze which cannot be opted out.
* Tentative suport for US Symbolic Layout (from X11 xkb/symbols/us) which provides greek letters, mathematical and typographical symbols. This task in incomplete at this point.
* Tested on Linux and Windows 10.
* Tested with TESmart KVM switch, which imposes latencies and presents issues, but it is still usable and fully functional.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idobo/keymaps/hyena/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 MechMerlin
```
Comment (fauxpark) [inline keyboards/idobo/keymaps/hyena/config.h:1]: This file can be deleted.

File: keyboards/idobo/keymaps/hyena/info.json [hunk 2]
```diff
@@ -0,0 +1,319 @@
+{
```
Comment (fauxpark) [inline keyboards/idobo/keymaps/hyena/info.json:1]: Use keymap.c or keymap.json, not both.

File: keyboards/idobo/keymaps/hyena/readme.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+pip3 install qmk
+
+# Ask QMK to setup the development environment for you:
+qmk setup
+
+# Build Hyena keymap and flash it onto your device:
+cd ~/qmk_firmware/keyboards/idobo/keymaps/hyena
+qmk json2c -o keymap.c info.json && \
+  qmk flash --keyboard idobo --keymap hyena
+```
```
Comment (fauxpark) [inline keyboards/idobo/keymaps/hyena/readme.md:22]:
```suggestion
    qmk flash -kb idobo -km hyena
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/idobo/keymaps/hyena/config.h (hunk 1 above)
Comment (frgomes) [reply keyboards/idobo/keymaps/hyena/config.h:1]: info.json was removed.

Comment (frgomes) [reply keyboards/idobo/keymaps/hyena/config.h:1]: I'm still working on two layers which involve Unicode. I will review the need of `config.h` when I finish the implementation.

File: keyboards/idobo/keymaps/hyena/info.json (hunk 2 above)
Comment (frgomes) [reply keyboards/idobo/keymaps/hyena/info.json]: .json file is now removed.

**Pull request closed without merging**